### ISSUE TITLE
chmod command in hydra build getting permission errors

### ISSUE THREAD

**Issue by Proph151Music** [Constellation-Labs/euclid-development-environment#4]
It looks like the hydra file is getting these errors because it is running the code without sudo, maybe?
When I manually run this within the euclid sdk directory it seems to unlock the permissions of the infra directory. But even though the permissions are fixed hydra always try's and it will always get the "Operation not permitted" error message.

```
scripts/hydra build --no_cache --only monitoring

\nCreating docker custom-network...

Network created

chmod: changing permissions of 'infra/docker/shared': Operation not permitted

chmod: changing permissions of 'infra/docker/shared/jars': Operation not permitted

chmod: changing permissions of 'infra/docker/shared/jars/cl-wallet.jar': Operation not permitted

chmod: changing permissions of 'infra/docker/shared/jars/currency-l1.jar': Operation not permitted

chmod: changing permissions of 'infra/docker/shared/jars/currency-l0.jar': Operation not permitted

chmod: changing permissions of 'infra/docker/shared/jars/global-l0.jar': Operation not permitted

chmod: changing permissions of 'infra/docker/shared/jars/cl-keytool.jar': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/prometheus/monitoring': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/prometheus/monitoring/prometheus': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/prometheus/monitoring/prometheus/data': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/prometheus/monitoring/prometheus/data/chunks_head': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/prometheus/monitoring/prometheus/data/lock': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/prometheus/monitoring/prometheus/data/queries.active': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/prometheus/monitoring/prometheus/data/wal': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/prometheus/monitoring/prometheus/data/wal/00000000': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/grafana/config': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/grafana/config/datasources': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/grafana/config/dashboards': Operation not permitted

chmod: changing permissions of 'infra/docker/monitoring/grafana/storage': Operation not permitted

chmod: changing permissions of 'source/global-l0/tessellation': Operation not permitted
```

**Comment by IPadawans** [comment]
Hey @Proph151Music, we deployed one fix on the branch `develop`. Could you please test if this solves your issue?

**Comment by Proph151Music** [comment]
I sure can.  I will test this a little later today and let you know.  It wasn't causing me an issue in particular... since I manually ensured the permissions were correct.  But I noticed it was throwing those errors regardless.

**Comment by Proph151Music** [comment]
It looks like it's working perfectly now. :)
My build is so far building perfectly on it's own, even though I did not use the chmod command manually.

**Comment by IPadawans** [comment]
Great, I'll close the issue then :) thanks!

**Comment by Proph151Music** [comment]
Actually... I'm not sure if this is related or not...
It seems grafana is not working after the build.  

hydra build --no_cache:
```Building monitoring image... (NO CACHE)

prometheus uses an image, skipping

**grafana uses an image, skipping**

monitoring image built

Cleaning up dangling docker images```

hydra start_genesis:
```Starting monitoring image ...

prometheus is up-to-date

Starting grafana ... done

monitoring image started```

hydra status:
```NAMES           PORTS                                                                                                                             STATUS

l1-currency-2   0.0.0.0:9300->9000/tcp, :::9300->9000/tcp, 0.0.0.0:9301->9001/tcp, :::9301->9001/tcp, 0.0.0.0:9302->9002/tcp, :::9302->9002/tcp   Up 2 minutes (healthy)

l1-currency-3   0.0.0.0:9400->9000/tcp, :::9400->9000/tcp, 0.0.0.0:9401->9001/tcp, :::9401->9001/tcp, 0.0.0.0:9402->9002/tcp, :::9402->9002/tcp   Up 2 minutes (healthy)

l1-currency-1   0.0.0.0:9200->9000/tcp, :::9200->9000/tcp, 0.0.0.0:9201->9001/tcp, :::9201->9001/tcp, 0.0.0.0:9202->9002/tcp, :::9202->9002/tcp   Up 3 minutes (healthy)

l0-currency     0.0.0.0:9100->9000/tcp, :::9100->9000/tcp, 0.0.0.0:9101->9001/tcp, :::9101->9001/tcp                                              Up 3 minutes (healthy)

l0-global       0.0.0.0:9000-9001->9000-9001/tcp, :::9000-9001->9000-9001/tcp                                                                     Up 4 minutes (healthy)

prometheus      0.0.0.0:9090->9090/tcp, :::9090->9090/tcp                                                                                         Up 4 hours```

**Comment by Proph151Music** [comment]
hydra start_genesis:
```Starting monitoring image ...

prometheus is up-to-date

Starting grafana ... done

monitoring image started```

hydra status:
```NAMES           PORTS                                                                                                                             STATUS

l1-currency-2   0.0.0.0:9300->9000/tcp, :::9300->9000/tcp, 0.0.0.0:9301->9001/tcp, :::9301->9001/tcp, 0.0.0.0:9302->9002/tcp, :::9302->9002/tcp   Up 2 minutes (healthy)

l1-currency-3   0.0.0.0:9400->9000/tcp, :::9400->9000/tcp, 0.0.0.0:9401->9001/tcp, :::9401->9001/tcp, 0.0.0.0:9402->9002/tcp, :::9402->9002/tcp   Up 2 minutes (healthy)

l1-currency-1   0.0.0.0:9200->9000/tcp, :::9200->9000/tcp, 0.0.0.0:9201->9001/tcp, :::9201->9001/tcp, 0.0.0.0:9202->9002/tcp, :::9202->9002/tcp   Up 3 minutes (healthy)

l0-currency     0.0.0.0:9100->9000/tcp, :::9100->9000/tcp, 0.0.0.0:9101->9001/tcp, :::9101->9001/tcp                                              Up 3 minutes (healthy)

l0-global       0.0.0.0:9000-9001->9000-9001/tcp, :::9000-9001->9000-9001/tcp                                                                     Up 4 minutes (healthy)

prometheus      0.0.0.0:9090->9090/tcp, :::9090->9090/tcp                                                                                         Up 4 hours```

**Comment by Proph151Music** [comment]
hydra status:
```NAMES           PORTS                                                                                                                             STATUS

l1-currency-2   0.0.0.0:9300->9000/tcp, :::9300->9000/tcp, 0.0.0.0:9301->9001/tcp, :::9301->9001/tcp, 0.0.0.0:9302->9002/tcp, :::9302->9002/tcp   Up 2 minutes (healthy)

l1-currency-3   0.0.0.0:9400->9000/tcp, :::9400->9000/tcp, 0.0.0.0:9401->9001/tcp, :::9401->9001/tcp, 0.0.0.0:9402->9002/tcp, :::9402->9002/tcp   Up 2 minutes (healthy)

l1-currency-1   0.0.0.0:9200->9000/tcp, :::9200->9000/tcp, 0.0.0.0:9201->9001/tcp, :::9201->9001/tcp, 0.0.0.0:9202->9002/tcp, :::9202->9002/tcp   Up 3 minutes (healthy)

l0-currency     0.0.0.0:9100->9000/tcp, :::9100->9000/tcp, 0.0.0.0:9101->9001/tcp, :::9101->9001/tcp                                              Up 3 minutes (healthy)

l0-global       0.0.0.0:9000-9001->9000-9001/tcp, :::9000-9001->9000-9001/tcp                                                                     Up 4 minutes (healthy)

prometheus      0.0.0.0:9090->9090/tcp, :::9090->9090/tcp                                                                                         Up 4 hours```

**Comment by Proph151Music** [comment]
It looks like almost everything in the infra folder has 777 access.  Except for the "infra/docker/shared" folder.

**Comment by Proph151Music** [comment]
After I changed the permissions on the shared folder to 777 and re-ran the build and start_genesis commands grafana works.
So the shared folder is being created with limited permissions and prevents grafana from properly working.

**Comment by Proph151Music** [comment]
After running the developer version it has the same issue.  Grafana still does not work.  The shared folder is created after the build/start_genesis commands but the folder has limited permissions which causes grafana not to work.

After I chmod to 777.  I do not need to run the hydra build command.  I just run the hydra start_genesis command and it successfully boots grafana up.

**Comment by IPadawans** [comment]
@Proph151Music Is expected that Grafana on build returns this:
```
prometheus uses an image, skipping

grafana uses an image, skipping

```
Grafana and Prometheus are simple images that are started only when you call the start_genesis command.

**Comment by IPadawans** [comment]
The shared folder should not impact the behavior of Grafana, this folder is used only to store some JARS. The Grafana will only start on the start_genesis command. The build command will skip the Grafana and Prometheus

**Comment by Proph151Music** [comment]
Yes that is what I am seeing.  

I don't think the shared folder is created during the build.  I don't recall seeing it this time until after the start_genesis command.  But the shared folder was created with limited access and grafana did not work until I performed a chmod 777 on the shared folder.

**Comment by IPadawans** [comment]
The shared folder is really created during the start_genesis command. It's weird, this folder shouldn't impact the Grafana 🤔 Let me investigate a little bit more on this

**Comment by Proph151Music** [comment]
OK. Thanks!  I'm fairly positive it somehow causes the issue.  Grafana would not work until I 777'd that shared folder.

**Comment by Proph151Music** [comment]
I ran a couple of tests on the develop branch pertaining to this.

If I do the sudo chmod command given to me to do before the hydra build everything does seem to work and grafana opens fine.

I did another test without doing the sudo chmod command and grafana does not work.  But it wasn't the shared folder that was blocking it this time.  It was the "infra/docker/monitoring/grafana/storage" folder that has limited access.  Once I changed the permissions on that exact folder I was able to run hydra start_genesis --only monitoring and then grafana worked.

I did also notice the "infra/docker/monitoring/grafana/config" folder had limited access as well.  But when I first tried to change the permissions on this and then try grafana it did not load until I changed the permissions on the storage folder.

**Comment by Proph151Music** [comment]
Upon a third look... It seems these folders aren't created during the hydra install.  So even if I run the sudo chmod command before hydra build it won't help.

After the hydra build command I still do not see those folders created.

It wasn't until after the hydra start_genesis command finalized that I saw those folders created.

Grafana would not open after hydra start_genesis. So I used "sudo chmod -R 777 storage" on the storage folder:
"infra/docker/monitoring/grafana/storage"

I noticed the storage folder was empty.
I then ran:
scripts/hydra start_genesis --only monitoring

Once it finished grafana was now working.

I noticed new folders and a grafana.db file were now created in the storage folder.

I did not change the permissions that were set on the config folder at all. It is still limited.
"infra/docker/monitoring/grafana/config"

I wonder if we create the "infra/docker/monitoring/grafana/storage" folder during the hydra install if it would prevent the permissions from being limited later when start_genesis runs?

**Comment by Proph151Music** [comment]
I did successfully test creating "infra/docker/monitoring/grafana/storage" at an early stage, well before hydra start_genesis.
Since the directory was already created with no limits on permissions grafana was fully functioning in the end as intended. :)

**Comment by IPadawans** [comment]
Hey @Proph151Music, thanks for testing and letting us know what you have done. We pushed a new version to the development branch that should cover this scenario. Could you please take a look if the problem was solved and let us know?

**Comment by Proph151Music** [comment]
I will test this build soon.  I sent a pull request last night as well.  I actually think we can get rid of a lot of the chmod commands.  I narrowed it down to the storage folder. It isn't created until toward the end. So the chmod commands on a lot of this won't affect it.

But by adding the empty storage folder in the GitHub and adding a chmod on the monitoring folder during hydra install ensures Grafana loads correctly. 😉

I also added an important fix for docker compose V2.
